### PR TITLE
fix(cts): split requests and e2e tests

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-2573](https://algolia.atlassian.net/browse/DI-2573)

Split e2e to be able to retry them, or exclude them when running the test suite offline.

[DI-2573]: https://algolia.atlassian.net/browse/DI-2573?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by shortcuts — COMMENTED** [review]
could you also add this condition in order to exclude them from forks? this way contributions can run and be merged https://github.com/algolia/api-clients-automation/blob/476c4ed565b5293202245f83d3cdf13d4653076f/.github/workflows/check.yml#L24

**Comment by millotp** [comment]
> could you also add this condition in order to exclude them from forks? this way contributions can run and be merged

yes good idea, I will make all the changes to the CI in the next PR, this is just to split them and generate all the files

File: scripts/ci/githubActions/createMatrix.ts [hunk 1]
```diff
@@ -69,15 +69,15 @@ async function createClientMatrix(baseBranch: string): Promise<void> {
...
+    const testsToDelete = `${testsOutputBase}/client ${testsOutputBase}/requests ${testsOutputBase}/requests_e2e`;
 
     // We only store tests of clients that ran during this job, the rest stay as is
     let testsToStore = matrix[language].toRun
       .map((client) => {
         const clientName = createClientName(client, language);
         const extension = getTestExtension(language);
 
-        return `${testsOutputBase}/client/${clientName}${extension} ${testsOutputBase}/requests/${clientName}${extension}`;
+        return `${testsOutputBase}/client/${clientName}${extension} ${testsOutputBase}/requests/${clientName}${extension} ${testsOutputBase}/requests_e2e/${clientName}${extension}`;
```
Comment (shortcuts) [inline scripts/ci/githubActions/createMatrix.ts:80]: maybe names should just be unit and e2e instead or requests

**Review by shortcuts — DISMISSED** [review]
I did not reviewed the mustache changes, in CI we trust

this is great!!

File: scripts/ci/githubActions/createMatrix.ts (hunk 1 above)
Comment (millotp) [reply scripts/ci/githubActions/createMatrix.ts:80]: thats a big change, its the good time to do it, but I feel like both `client` and `request` tests are unit tests, maybe  I can just rename `request_e2e` to `responses` or `e2e`

**Review by shortcuts — APPROVED** [review]
Wonderful execution

**Pull request merged**